packages/jaeger-ui-components/src/TracePageHeader/SpanGraph/index.test.tsx

```diff
@@ -39,52 +37,35 @@ describe('<SpanGraph>', () => {
-
   beforeEach(() => {
-    wrapper = shallow(<SpanGraph {...(props as unknown as SpanGraphProps)} />);
-  });
-
-  it('renders a <CanvasSpanGraph />', () => {
-    expect(wrapper.find(CanvasSpanGraph).length).toBe(1);
+    render(<SpanGraph {...(props as unknown as SpanGraphProps)} />);
   });
 
-  it('renders a <TickLabels />', () => {
-    expect(wrapper.find(TickLabels).length).toBe(1);
+  it('renders <CanvasSpanGraph />', () => {
+    const canvasSpanGraphComponent = screen.getByTestId('CanvasSpanGraph');
+    expect(canvasSpanGraphComponent).toBeTruthy();
   });
 
-  it('returns a <div> if a trace is not provided', () => {
-    wrapper = shallow(<SpanGraph {...({ ...props, trace: null } as unknown as SpanGraphProps)} />);
-    expect(wrapper.matchesElement(<div />)).toBeTruthy();
+  it('renders <TickLabels />', () => {
+    const tickLabelsComponent = screen.getByTestId('TickLabels');
+    expect(tickLabelsComponent).toBeTruthy();
   });
 
-  it('passes the number of ticks to render to components', () => {
-    const tickHeader = wrapper.find(TickLabels);
-    const viewingLayer = wrapper.find(ViewingLayer);
-    expect(tickHeader.prop('numTicks')).toBeGreaterThan(1);
-    expect(viewingLayer.prop('numTicks')).toBeGreaterThan(1);
-    expect(tickHeader.prop('numTicks')).toBe(viewingLayer.prop('numTicks'));
+  it('returns an empty div if a trace is not provided', () => {
+    const { container } = render(<SpanGraph {...({ ...props, trace: null } as unknown as SpanGraphProps)} />);
+    expect(container.firstChild).toBeEmptyDOMElement();
   });
 
-  it('passes items to CanvasSpanGraph', () => {
-    const canvasGraph = wrapper.find(CanvasSpanGraph).first();
-    const items = trace?.spans.map((span) => ({
-      valueOffset: span.relativeStartTime,
-      valueWidth: span.duration,
-      serviceName: span.process.serviceName,
-    }));
-    expect(canvasGraph.prop('items')).toEqual(items);
+  it('renders <TickLabels /> with the correct numnber of ticks', async () => {
+    const tickLabelsDiv = screen.getByTestId('TickLabels');
+    expect(getAllByTestId(tickLabelsDiv, 'tick').length).toBe(TIMELINE_TICK_INTERVAL + 1);
   });
 
-  it('does not regenerate CanvasSpanGraph without new trace', () => {
-    const canvasGraph = wrapper.find(CanvasSpanGraph).first();
-    const items = canvasGraph.prop('items');
-
-    wrapper.instance().forceUpdate();
-
-    const newCanvasGraph = wrapper.find(CanvasSpanGraph).first();
-    const newItems = newCanvasGraph.prop('items');
-
-    expect(newItems).toBe(items);
+  it('renders <TickLabels /> with the correct value next to each tick', async () => {
+    render(<TickLabels numTicks={5} duration={1} />);
+    expect(await screen.findAllByText(/0.2/)).toBeTruthy();
+    expect(await screen.findAllByText(/0.4/)).toBeTruthy();
+    expect(await screen.findAllByText(/0.6/)).toBeTruthy();
+    expect(await screen.findAllByText(/0.8/)).toBeTruthy();
```

Comment:
You can use other selectors depending on what you need. For example here you could probably use `getBy...` here if you expect only single element and there probably isn't any async rendering.

Reply:
I've changed the selector to `getByText` in commit https://github.com/grafana/grafana/pull/60200/commits/7f0413aaafcc41240bcf053e6c13a1bb6976c378.